caesar_cipher.rb

```diff
@@ -1,25 +1,24 @@
+
+def get_shifted_idx(letters_array, char, shift)
+    shifted_idx = letters_array.index(char) + shift
+  if shifted_idx > letters_array.size
+    shifted_idx -= letters_array.size
+  end
+  return shifted_idx
+end
+
 def caesar_cipher(str, shift) 
-    lowercase_letters = Array('a'..'z')
-    uppercase_letters = Array('A'..'Z')
-    new_str = ''
-    str.each_char do |c|
-        if lowercase_letters.include?(c)
-            shifted_idx = lowercase_letters.index(c) + shift
-            # if the new idx is out of bounds, loop back to beginning for idx
-            # take idx value and subtract from 26 to get new idx value
-            if shifted_idx > 26
-                shifted_idx -= 26
-            end
-            new_str += lowercase_letters[shifted_idx]
-        elsif uppercase_letters.include?(c)
-            shifted_idx = uppercase_letters.index(c) + shift
-            if shifted_idx > 26
-                shifted_idx -= 26
-            end
-            new_str += uppercase_letters[shifted_idx]       
-        else
-            new_str += c       
-        end
+  cc_str = ''
+  lowercase_letters = ('a'..'z').to_a
+  uppercase_letters = ('A'..'Z').to_a
+  str.split("").each_with_object("") do |c| 
+    if lowercase_letters.include?(c)
+      cc_str << lowercase_letters[get_shifted_idx(lowercase_letters, c, shift)]
+    elsif uppercase_letters.include?(c)
+      cc_str << uppercase_letters[get_shifted_idx(uppercase_letters, c, shift)]
+    else
+      cc_str << c
     end
-    return new_str
+  end
+  return cc_str
```

Comment:
And I think you can remove this also. The result of `str.split("").each_with_object("") do |c, cc_str|` will be the `cc_str`. And, you have good tests, so you can make sure I'm not lying to you.

Reply:
😱 magic!